base/libgit2/callbacks.jl

```diff
@@ -187,8 +187,10 @@ function authenticate_userpass(libgit2credptr::Ptr{Ptr{Void}}, p::CredentialPayl
     if p.use_git_helpers && (!revised || !isfilled(cred))
         git_cred = GitCredential(p.config, p.url)
 
-        cred.user = Base.get(git_cred.username, "")
-        cred.pass = Base.get(git_cred.password, "")
+        # Use `deepcopy` to ensure zeroing the `git_cred` doesn't also zero the `cred`s copy
+        cred.user = deepcopy(Base.get(git_cred.username, ""))
+        cred.pass = deepcopy(Base.get(git_cred.password, ""))
+        securezero!(git_cred)
         revised = true
```

Comment:
It seems like we should also call `securezero!(cred)` at the end of the `authenticate_userpass` function.   The function returns the result of `git_cred_userpass_plaintext_new`, but this function [makes its own copy of the password](https://libgit2.github.com/libgit2/#HEAD/group/cred/git_cred_userpass_plaintext_new).

Reply:
We can't run `securezero!(cred)` at the end of `authenticate_userpass` as the credential information is still needed to be used in the [`approve(::CredentialPayload)`](https://github.com/JuliaLang/julia/blob/6fa2047f1af1ad6d752302cb2992ad850a90718b/base/libgit2/types.jl#L1326) and the [`reject(::CredentialPayload)`](https://github.com/JuliaLang/julia/blob/6fa2047f1af1ad6d752302cb2992ad850a90718b/base/libgit2/types.jl#L1344) methods. We should be able to run `securezero!` on the credentials at the end of the approve/reject methods but I'll need to add in some extra logic to skip this when running the test cases.

I was planning to do this in a future PR.

Reply:
Oh, I see, `cred` is actually still available via `get(p.credential)`.